src/evolutionaryAlgorithm/EvolutionSimulator.py

```diff
@@ -34,43 +36,52 @@ def fromSimulationComponentList(SimulationComponents):
         mutator = SimulationComponents["Mutator"]
         parentSelector = SimulationComponents["ParentSelector"]
         survivorSelector = SimulationComponents["SurvivorSelector"]
+        fitnessFunction = SimulationComponents["FitnessFunction"]
 
-        return EvolutionSimulator(initializator, recombinator, mutator, parentSelector, survivorSelector)
+        return EvolutionSimulator(initializator, recombinator, mutator, parentSelector, survivorSelector,
+                                  fitnessFunction)
 
     def run(self):
         population = self.initialize_population()
+        self.markPopulation(population)
+        # marriages = self.marry(population)
 
         while (not self.endSimulation()):
             parents = self.selectParents(population)
             offspring = self.applyRecombinations(parents)
             mutated = self.applyMutations(offspring)
-            newGenerationCandidats = self.selectNewGeneration(offspring, population)
-            population = self.selectPopulation(newGenerationCandidats)
+            self.markPopulation(mutated)
```

Comment:
co to jest za zmienna mark population?

Reply:
a w sensie mark znaczy evaluate? Musimy to zmienić bo nie będzie zmieniana ewaluacja

Reply:
tak, chodziło o evaluate